### PR TITLE
Get Robot building and working again.

### PR DESCRIPTION
**Pull request by pmuetschard**

### CODE REVIEW

File: cmd/gapit/trace.go [hunk 1]
```diff
@@ -99,12 +96,32 @@ func (verb *traceVerb) Run(ctx context.Context, flags flag.FlagSet) error {
...
+				}
+			}
+		} else if len(devices) == 1 && strings.HasPrefix(traceURI, "apk:") {
+			data, err := ioutil.ReadFile(traceURI[4:])
+			if err != nil {
+				return log.Errf(ctx, err, "Failed to read APK at %s", traceURI[4:])
+			}
+			target = func(opts *service.TraceOptions) {
+				opts.Device = devices[0]
+				opts.App = &service.TraceOptions_UploadApplication{
```
Comment (pau-baiget) [inline cmd/gapit/trace.go]: Does this take care if the apk is already installed?

File: cmd/gapit/trace.go [hunk 2]
```diff
@@ -99,12 +96,32 @@ func (verb *traceVerb) Run(ctx context.Context, flags flag.FlagSet) error {
...
+					data,
```
Comment (ben-clayton) [inline cmd/gapit/trace.go]: Go style requires field names for initializers of structs declared in another package.

File: cmd/gapit/trace.go [hunk 3]
```diff
@@ -137,65 +154,59 @@ func (verb *traceVerb) Run(ctx context.Context, flags flag.FlagSet) error {
...
-		fmt.Printf("Tracing %+v\n", found[0].uri)
-		out = found[0].name + ".gfxtrace"
-		uri = found[0].uri
-		traceDevice = found[0].device
+			fmt.Printf("Tracing %+v\n", found[0].uri)
+			out = found[0].name + ".gfxtrace"
+			target = func(opts *service.TraceOptions) {
+				opts.Device = found[0].device
+				opts.App = &service.TraceOptions_Uri{
+					found[0].uri,
```
Comment (ben-clayton) [inline cmd/gapit/trace.go]: Ditto

File: gapis/trace/android/trace.go [hunk 4]
```diff
@@ -346,6 +346,10 @@ func (t *androidTracer) InstallPackage(ctx context.Context, o *service.TraceOpti
 			pkg.Uninstall(ctx)
 		}
 	}
+
+	o.App = &service.TraceOptions_Uri{
+		info.URI(),
```
Comment (ben-clayton) [inline gapis/trace/android/trace.go]: And again

File: cmd/gapit/trace.go (hunk 1 above)
Comment (AWoloszyn) [reply cmd/gapit/trace.go]: Following this through to here:
https://github.com/google/gapid/blob/master/gapis/trace/android/trace.go#L319

We reinstall the APK if it is already installed (and grant it all permissions)

**Comment by AWoloszyn** [comment]
Other than style nits, the only thing I want to comment on is that I think how we package up OBB files for Robot will have changed.

Now it is expected that you EITHER pass a .apk file down in this call OR a zip file that contains your your .apk + .obb files. This way everything can be in a single file.

File: cmd/gapit/trace.go (hunk 2 above)
Comment (pmuetschard) [reply cmd/gapit/trace.go]: Done.

File: cmd/gapit/trace.go (hunk 3 above)
Comment (pmuetschard) [reply cmd/gapit/trace.go]: Done.

File: gapis/trace/android/trace.go (hunk 4 above)
Comment (pmuetschard) [reply gapis/trace/android/trace.go]: Done.

**Comment by pmuetschard** [comment]
re: obb - if put in a TODO and made robot return an error if a subject with an OBB is attempted to be traced. If we actually want OBBs again we can fix robot to properly package them, or fix gapit to do it for robot.

**Pull request merged**